File: python/ajustado/src/app/modules/auth/dtos/outputs/login_auth_clients_dto.py
from pydantic import BaseModel, ConfigDict, field_serializer
from datetime import datetime

from app.modules.clients.entity import ClientEntity

class LoginAuthClientsOutputDto(BaseModel):
    token: str
    usuario: ClientEntity
    expiraEm: datetime

    model_config = ConfigDict(serialization={'datetime_format': 'iso8601'})

    def to_json(self):
        return self.model_dump()

    @field_serializer("usuario")
    def serialize_usuario(self, usuario: ClientEntity):
        return usuario.to_json()

    @field_serializer("expiraEm")
    def serialize_expira_em(self, value: datetime):
        """Garante que o datetime seja convertido para ISO 8601"""
        return value.isoformat()

File: typescript/express_orm/src/app/modules/auth/dtos/outputs/login-auth-clients.dto.ts
import { Transform, instanceToPlain } from "class-transformer";

// Entities
import { ClientEntity } from "@/app/modules/clients/entities/client.entity";

export class LoginAuthClientsOutputDto {
  @Transform(({ value }) => value.toJSON())
  usuario: Partial<ClientEntity>;

  token: string;
  expiraEm: string;

  constructor(partial: Partial<LoginAuthClientsOutputDto>) {
    Object.assign(this, partial);
  }

  toJSON() {
    return instanceToPlain(this);
  }
}
